### PR TITLE
Aamc methods

### PR DESCRIPTION
**Pull request by stopfstedt**
supersedes #3033

### CODE REVIEW

**Comment by jrjohnson** [comment]
Deploy preview ready!

Built with commit 95c0efe8c8a7770eea3cca2c4c94b93a30c13315

https://deploy-preview-3047--ilios-frontend.netlify.com

**Comment by saschaben** [comment]
still needs to filter AAMC method options on edit of existing types, so that an assessment method cannot map to an instructional, and vice versa.

**Comment by jrjohnson** [comment]
Still awaiting tests - question for @saschaben I made the AAMCMethod not-required (can be blank). Is that right? This can be click tested again now.

I also prevented aamcMethod from being changed if there are attached sessions.

File: app/components/school-session-type-form.js [hunk 1]
```diff
@@ -19,22 +20,57 @@ const Validations = buildValidations({
...
 
 export default Component.extend(ValidationErrorDisplay, Validations, {
   store: service(),
   classNames: ['school-session-type-form'],
   title: null,
+  selectedAamcMethodId: null,
   calendarColor: null,
   assessment: null,
   assessmentOptionId: null,
+  aamcMethods: computed('assessment', async function(){
```
Comment (jrjohnson) [inline app/components/school-session-type-form.js]: Performance can be improved here by splitting this into an allAaamcMethods property and a filtered aamcMethids property.

**Comment by saschaben** [comment]
`I made the AAMCMethod not-required (can be blank). Is that right?` THIS IS CORRECT.

`I also prevented aamcMethod from being changed if there are attached sessions.` THIS IS INCORRECT: A MAPPING MAY BE CHANGED, EVEN AFTER SESSIONS ARE ATTACHED.
>> The Rule: Once created&saved, a session type may only change type within its class -- either instructional or assessment. Once created&saved, an assessment type may not be changed from summative /formative selection. Once _mapped_, a session type _MAY_ its mapping changed, but only within its class (instructional/assessment).

Currently hitting a few errors on the form:

For an unassociated session type (no sessions in use), the following fields are shown as modifiable, but ALL throw error:
preexist2nomap

err1

we should lock down all form fields according to the rules described above. The assessment toggle should not be active after save. The AAMC Method select list should be constrained to the class selected, and active. If assessment, the formative/summative selector should be inactive after save.

preexist1

**Comment by saschaben** [comment]
Looks good. Approved from this end. :+1:

**Pull request merged**